readthedocs/restapi/views/integrations.py

```diff
@@ -191,7 +212,13 @@ def handle_webhook(self):
         # Handle push events and trigger builds
         if event == GITLAB_PUSH:
             try:
-                branches = [self.request.data['ref'].replace('refs/heads/', '')]
+                # GitLab only returns one branch.
+                branch = {
+                    'name': self.request.data['ref'].replace('refs/heads/', ''),
```

Comment:
Won't this interferes with your other PR at #3546?

Reply:
Just when merging/rebasing, on the logic everything would work as expected :)